quill-core/src/main/scala/io/getquill/quotation/Quotation.scala

```diff
@@ -9,45 +9,41 @@ import io.getquill.ast._
 import scala.reflect.internal.Symbols
 
 import scala.collection.mutable
+import scala.reflect.NameTransformer
 
 trait Quoted[+T] {
   def ast: Ast
 }
 
 case class QuotedAst(ast: Ast) extends StaticAnnotation
 
-trait Quotation extends Liftables with Unliftables {
+trait Quotation extends Liftables with Unliftables with Parsing {
 
   val c: Context
   import c.universe._
 
   def quote[T: WeakTypeTag](body: Expr[T]) = {
 
-    val freeVars = extractFreeVars(body.tree)
-
-    val ast = Parsing(c)(freeVars).astParser(body.tree)
+    val ast = astParser(body.tree)
+    
+    def bindingName(s: String) =
+      TermName(NameTransformer.encode(s))
```

Comment:
I really like this approach. Just don't forget that we might get a collusion with the example below:
```scala
val q1 = {
    val temp = 1
    quote { q.filter(_.i == lift(temp))}
}
val q2 = {
    val temp = 2L
    quote { q.filter(_.l == lift(temp)) }
}
db.run(q2)
```